### ISSUE TITLE
Described generalises fixed point iteration in docs

### ISSUE THREAD

**Issue by ddundo** [mesh-adaptation/mesh-adaptation-docs#29]
In order to clarify the on-the-fly adaptation approach (for which I will soon open a PR in Goalie), we should describe the generalised fixed point iteration described in https://www.sciencedirect.com/science/article/abs/pii/S002199910600369X - and perhaps even take Figure 3 from it.

**Comment by jwallwork23** [comment]
Yeah we need to do a docs overhaul. Probably a good idea to have another docs sprint soon.
